journalpump/journalpump.py

```diff
@@ -675,6 +725,33 @@ def _truncate_long_message(self, json_entry):
         }
         return json.dumps(entry, default=default_json_serialization).encode("utf8")
 
+    def _apply_secret_filters(self, data):
+        changed = False
+
+        for secret_filter in self.reader.secret_filters:
+            msg = data.get("MESSAGE")
+            if msg:
+                match = secret_filter["compiled_pattern"].match(msg)
+                if match:
+                    msg = ""
+                    changed = True
+
+                    if self.reader.secret_filter_metrics is True:
+                        self.pump.stats.increase(
```

Comment:
I presume matches are generally speaking rare but if for whatever reason a lot of the log lines match the pattern this could result in a lot of statsd messages getting sent. Would seem a bit safer to just increase a counter here and have the main loop send out the value.

Reply:
good call, fixed